contracts/src/libs/CoreLib.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GNU GPLv3
+pragma solidity ^0.8.29;
+
+import "../interfaces/Constants.sol";
+import "../interfaces/Types.sol";
+import "../interfaces/Errors.sol";
+import "../interfaces/Events.sol";
+
+library CoreLib {
+    // ------------------------------------------------------------------
+    // Internal functions
+    // ------------------------------------------------------------------
+
+    /**
+     * @notice Computes the unique EIP-712 digest for a SafeTx using the provided parameters and domain.
+     * @param safeAddress Address of the target Safe Smart Account.
+     * @param chainId Chain ID included in the domain separator.
+     * @param nonce Safe transaction nonce.
+     * @param to Target address the Safe will call.
+     * @param value ETH value to be sent with the call.
+     * @param data Call data executed by the Safe.
+     * @param operation Operation type: 0 = CALL, 1 = DELEGATECALL.
+     * @param safeTxGas Gas limit for the Safe's internal execution.
+     * @param baseGas Base gas overhead for reimbursement.
+     * @param gasPrice Gas price used for reimbursement calculation.
+     * @param gasToken Token address for refunds (0x0 for ETH).
+     * @param refundReceiver Address to receive gas refunds.
+     * @return safeTxHash Keccak256 digest of the EIP-712 encoded SafeTx.
+     */
+    function _computeSafeTxHash(
+        address safeAddress,
+        uint256 chainId,
+        uint256 nonce,
+        address to,
+        uint256 value,
+        bytes memory data,
+        uint8 operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver
+    ) internal pure returns (bytes32 safeTxHash) {
+        bytes32 domainSeparator = keccak256(
+            abi.encode(_DOMAIN_TYPEHASH, chainId, safeAddress)
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(
+                _SAFE_TX_TYPEHASH,
+                to,
+                value,
+                keccak256(data),
+                operation,
+                safeTxGas,
+                baseGas,
+                gasPrice,
+                gasToken,
+                refundReceiver,
+                nonce
+            )
+        );
+        safeTxHash = keccak256(
+            abi.encodePacked("\x19\x01", domainSeparator, structHash)
+        );
+    }
+
+    /**
+     * @notice Splits a 65-byte ECDSA signature into its components and recovers the signer address.
```

Comment:
not for this PR, but since we are enforcing low-s signatures, we can encode the `v` value in `s` to save some call data - then signatures are `(r, vs)` pair instead of `bytes` (which adds one word of call data for the offset pointer, one word for the length witch MUST be 65, and then 3 words for actual bytes with padding - a 40% encoding efficiency).

Again, nothing to do here - just something that we can look into when gas optimisation time comes around (when exploring economic viability of sponsoring signature submission).